src/stdlib/dns_lookup.rs

```diff
@@ -34,9 +36,18 @@ mod non_wasm {
             .map_err(|err| format!("parsing query class failed: {err}"))?;
 
         let conf = build_options(options.try_object()?)?;
-        let answer = StubResolver::run_with_conf(conf, move |stub| async move {
-            stub.query((host, qtype, qclass)).await
-        })
+        let answer = match Handle::try_current() {
+            Ok(_) => thread::spawn(move || {
```

Comment:
I think if it returns `Ok` you can use something like:

```
futures::executor::block_on(async {
        handle
            .spawn(async {})
})
```

to just execute in the current Tokio runtime instead of spawning a separate thread per https://stackoverflow.com/a/62536772

Reply:
I still have to await on return value of `block_on` which bring us back to the same issue. I think we will have to go with a pool or a dedicated thread until proper async support is added to VRL (if it ever gets added, since it maybe clashes with original goals).

Reply:
Ah gotcha. Then yeah, a pool seems like a good approach. If/when VRL functions become async we can revisit this. I was going to suggest we thread the through the runtime from Vector into VRL, but I think I'd prefer to just make the functions async than do that.